src/main/java/io/confluent/kafkarest/v2/KafkaConsumerManager.java

```diff
@@ -519,21 +503,22 @@ public void run() {
         try {
           while (isRunning.get()) {
             long now = time.milliseconds();
-            while (!consumersByExpiration.isEmpty() && consumersByExpiration.peek().expired(now)) {
-              final KafkaConsumerState state = consumersByExpiration.remove();
-              consumers.remove(state.getId());
-              executor.submit(new Runnable() {
-                @Override
-                public void run() {
-                  state.close();
-                }
-              });
+            Iterator itr = consumers.values().iterator();
+            while (itr.hasNext()) {
+              final KafkaConsumerState state = (KafkaConsumerState) itr.next();
+              if (state.expired(now)) {
+                log.debug("Removing the expired consumer {}", state.getId());
+                itr.remove();
+                executor.submit(new Runnable() {
+                  @Override
+                  public void run() {
+                    state.close();
+                  }
+                });
+              }
             }
-            long timeout =
-                consumersByExpiration.isEmpty()
-                ? Long.MAX_VALUE
-                : consumersByExpiration.peek().untilExpiration(now);
-            KafkaConsumerManager.this.wait(timeout);
+
+            KafkaConsumerManager.this.wait(1000);
```

Comment:
Could we use `sleep` instead now that we don't require the expiration thread to be woken up explicitly?